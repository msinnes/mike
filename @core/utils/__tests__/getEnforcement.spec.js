const getEnforcement = require('@core/utils/getEnforcement');

describe('getEnforcement', () => {
  afterEach(() => jest.resetAllMocks);

  it('should be a function', () => {
    expect(getEnforcement).toBeDefined();
    expect(getEnforcement).toBeInstanceOf(Function);
  });

  it('should return a function', () => {
    const enforcement = getEnforcement();
    expect(enforcement).toBeDefined();
    expect(enforcement).toBeInstanceOf(Function);
  });

  it('should run a check function and throw an error if the check fails', () => {
    const checkMock = jest.fn();
    const ENFORCEMENT_STRING = 'string';
    const enforcement = getEnforcement(checkMock, ENFORCEMENT_STRING);
    checkMock.mockReturnValueOnce(true);
    expect(() => {
      enforcement();
    }).not.toThrow();

    checkMock.mockReturnValueOnce(false);
    expect(() => {
      enforcement();
    }).toThrowErrorMatchingSnapshot();
  });
});