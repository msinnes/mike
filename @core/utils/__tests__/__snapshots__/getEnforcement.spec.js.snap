// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEnforcement should run a check function and throw an error if the check fails 1`] = `"InternalError: string"`;
